src/Tickets/Blocks/Controller.php

```diff
@@ -100,14 +100,18 @@ public function register_for_blocks() {
 	 * Register the blocks after plugins are fully loaded.
 	 *
 	 * @since 5.3.0
+	 * @since TBD Correctly get post type when creating a new post or page.
 	 */
 	public function register_blocks() {
 		if ( is_admin() ) {
 			// In admin context, do not register the blocks if the post type is not ticketable.
-			$post      = get_post() ?: get_post( tribe_get_request_var( 'post' ) );
-			$post_type = get_post_type( $post ) ?: tribe_get_request_var( 'post_type' );
+			$post_id           = tribe_get_request_var( 'post' );
+			$post_type_default = 'post';
+
+			$post_type = $post_id ? get_post_type( get_post( $post_id ) ) : tribe_get_request_var( 'post_type', $post_type_default );
```

Comment:
This can be simplified:
```suggestion
			$post_type = $post_id ? get_post_type( $post_id ) : tribe_get_request_var( 'post_type', $post_type_default );
```